src/Parser/GuardianDruid/Modules/Items/Skysecs.js

```diff
@@ -0,0 +1,118 @@
+ * Skysec's Hold from the total healing done by Skysec's, and shows the "effective HPS" of the
+ * boots.
+ */
+class SkysecsHold extends Module {
+  currentCycleFRTicksLeft = 0;
+  currentCycleSHTicksLeft = 0;
+  potentialHeal = 0;
+  currentCycleEffectiveHeal = 0;
+  overallEffectiveHeal = 0;
+  lastSkysecsTick = 0;
+  totalCasts = 0;
+  averagePlayerHPs = [];
+  efficiency = 1;
+
+  /**
+   * FR healing with SH operates in cycles; two ticks of FR then one tick of SH, repeated 3 times.
+   * We only count FR overhealing against SH ticks from the same cycle, since damage and other healing
+   * can occur between FR casts (damage and healing can occur within the duration of an FR cast, but
+   * the time between ticks is short enough that I consider this to be an acceptable margin of error).
+   *
+   * FR can also be "double-tapped" by casting it again before all ticks of the previous cast have
+   * occurred.  We treat this as an extension of the original cycle, since the remaining FR healing
+   * from the first cast is folded into the second.  The heal from SH is not folded in the same way
+   * (it's simply reset to 12% of maxHP), so if this occurs we consider the remaining heal from
+   * the original cast as wasted.
+   */
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.FRENZIED_REGENERATION.id) {
+      this.totalCasts += 1;
+      this.averagePlayerHPs.push(event.maxHitPoints);
+      // FR always ticks 6 times, even if it's cast again while the current one is still
+      // ticking (the remaining heal is added to the new cast)
+      if (this.currentCycleFRTicksLeft === 0) {
+        this.overallEffectiveHeal += this.currentCycleEffectiveHeal;
+        this.currentCycleEffectiveHeal = 0;
+        this.potentialHeal += this.lastSkysecsTick * this.currentCycleSHTicksLeft;
+      }
+      this.currentCycleFRTicksLeft = 6;
+      this.currentCycleSHTicksLeft = 3;
+    }
+  }
+
+  on_byPlayer_heal(event) {
+    const overheal = event.overheal || 0;
+    if (event.ability.guid === SPELLS.SKYSECS_HOLD_HEAL.id) {
+      this.currentCycleEffectiveHeal += event.amount;
+      this.potentialHeal += event.amount + overheal;
+      this.lastSkysecsTick = event.amount + overheal;
+      this.currentCycleSHTicksLeft -= 1;
+    }
+
+    if (event.ability.guid === SPELLS.FRENZIED_REGENERATION.id) {
+      this.currentCycleEffectiveHeal = Math.max(this.currentCycleEffectiveHeal - overheal, 0);
+      this.currentCycleFRTicksLeft -= 1;
+    }
+  }
+
+  item() {
+    const effectiveHeal = this.overallEffectiveHeal + this.currentCycleEffectiveHeal;
+    this.efficiency = effectiveHeal / this.potentialHeal;
+
+    const fightLengthSec = (this.owner.fight.end_time - this.owner.fight.start_time) / 1000;
+
+    // Skysec's heal
+    const averagePlayerHP = this.averagePlayerHPs.reduce((sum, hp) => sum + hp, 0) / this.totalCasts;
+    const effectiveHealPerCast = this.efficiency * 0.12 * averagePlayerHP;
```

Comment:
wrt the 0.12; Please do not use any *magic numbers* in the code and define them as a constant at the top of your file if they're only going to be used in one place and you don't need any additional info (e.g. `const FRENZIEN_REGENERATION_TICKS_PER_CAST = 6;`).